### ISSUE TITLE
(!) Admin UI no longer compiles / starts after updating to version 2.2.7

### ISSUE THREAD

**Issue by KariNarhi28** [vendure-ecommerce/vendure#2943]
**Describe the bug**
I updated the Vendure-packages to 2.2.7 and tried to start the server.

Instead of starting up, I got this error:
```
Running yarn run build
yarn run v1.22.19
$ ng build --configuration production
Node packages may not be installed. Try installing with 'yarn install'.
Error: Could not find the '@angular-devkit/build-angular:dev-server' builder's node package.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
1
```
Also if I try to build the Admin UI (compile using the utility script), I get the same but with:
`Error: Could not find the '@angular-devkit/build-angular:browser' builder's node package.`

Clearly the issue is the missing @angular-devkit/build-angular node package, which I believe is part of the inner workings of the Admin UI.

**To Reproduce**
1. Make sure you have Vendure packages as version 2.2.7
2. Try to start up the Vendure API + Admin UI.
   
Here are my scripts for starting with dev-mode (worker included):
```json
"dev:server": "ts-node ./src/index.ts",
"dev:worker": "ts-node ./src/index-worker.ts",
"dev": "concurrently npm:dev:*"
```
3. Run `npm run dev` if you have same scripts
4. After the API has succesfully setup, the Admin UI starts to compile.
5. You should see the mentioned error.

**Expected behavior**
Start up and compile as it is supposed to.

**Environment (please complete the following information):**
 - @vendure/core version: 2.2.7
 - Nodejs version: v20.13.1
 - Database (mysql/postgres etc): SQLite

**Additional context**
I have no custom plugins integrated into the UI.

**Comment by KariNarhi28** [comment]
As far as I understand, this has something to do with the [Admin UI scaffold under ui-devkit](https://github.com/vendure-ecommerce/vendure/blob/master/packages/ui-devkit/scaffold).

Most likely when the dependencies are put into the compilation, the @angular-devkit/build-angular is not included.

**Comment by michaelbromley** [comment]
Hi,

I'm interested in this step:

> Upgraded the packages that came with it.

What version of which packages needed to be upgraded?

Also note that yarn v1.x is not supported. It is incompatible with some of our dependencies now, e.g. Sharp.

**Comment by KariNarhi28** [comment]
I don't use Yarn, so whatever related to it is what comes in the packages.

As for the packages I updated after initial creation of the project:

I used ncu (npm-check-updates) and got this result:
```
better-sqlite3  9.6.0  →  11.1.2
 typescript      5.3.3  →   5.5.3

Run ncu -u to upgrade package.json
```

So those 2 packages are not up-to-date after initial creation, and I updated them as ncu suggested.

**Comment by KariNarhi28** [comment]
OK. That was it.

So the actual issue was my eagerness to keep all packages updated.

So when I updated to 2.2.7 with ncu, the typescript-package also updated and that broke the system.

---

Thanks for the help, feeling really dumb now that I wasted 2 days because of something this simple.

Actually, I have another critical issue about persisting the login-state when refreshing page #2940, would really appreciate help with that too.

Closing the issue.
